### ISSUE TITLE
Inception weights or preprocessing issue

### ISSUE THREAD

**Issue by enizhibitsky** [fchollet/deep-learning-models#16]
I have a problem with using Inception for fine-tuning. Actually, there is either an issue with initalized Inception itself according to its predictions or the preprocessing script is not suitable for both ResNet and Inception models.

Running
```python
import numpy as np

from keras.preprocessing import image

from imagenet_utils import preprocess_input, decode_predictions
from resnet50 import ResNet50

model = ResNet50(weights='imagenet')

img_path = 'elephant.jpg'
img = image.load_img(img_path, target_size=(224, 224))
x = image.img_to_array(img)
x = np.expand_dims(x, axis=0)
x = preprocess_input(x)

preds = model.predict(x)
print('Predicted:', decode_predictions(preds))
```
gives me `African_elephant` with p=0.97174376, whilst running
```python
import numpy as np

from keras.preprocessing import image

from imagenet_utils import preprocess_input, decode_predictions
from inception_v3 import InceptionV3

model = InceptionV3(weights='imagenet')

img_path = 'elephant.jpg'
img = image.load_img(img_path, target_size=(299, 299))
x = image.img_to_array(img)
x = np.expand_dims(x, axis=0)
x = preprocess_input(x)

preds = model.predict(x)
print('Predicted:', decode_predictions(preds))
```
states that my elephant is a `suspension_bridge` with p=0.99999976.

I've already tried deleting keras cache and I use `tf` backend.

**Comment by apiszcz** [comment]
I am using this pattern with fine tuning and two output classes, observations: random classification output results with a single test image. Training was on 1 image for each class and indicated accuracy of 1.0 with an error less 1e-07.   Displaying the image after the img_to_array call is an inverted color image.

            image = image_utils.load_img(file, target_size=(299, 299))
            image = image_utils.img_to_array(image)
            image = np.expand_dims(image, axis=0)
            image = preprocess_input(image)
            preds = model.predict(image)
